datafusion/datasource-parquet/src/row_group_filter.rs

```diff
@@ -70,6 +78,49 @@ impl RowGroupAccessPlanFilter {
         self.access_plan
     }
 
+    /// Returns the is_fully_matched vector
+    pub fn is_fully_matched(&self) -> &Vec<bool> {
+        &self.is_fully_matched
+    }
+
+    /// Prunes the access plan based on the limit and fully contained row groups.
```

Comment:
I think we should explain the rationale here and leave a link to the paper -- I think you can adapt the very nice description on https://github.com/apache/datafusion/issues/18860

Basically:
1. Why is this optimization important: (can reduce the number of RowGroups needed with a known limit)
2. Why is it correct (the rationale on https://github.com/apache/datafusion/issues/18860)